engine/player/unique_gear_thewarwithin.cpp

```diff
@@ -6919,6 +6919,239 @@ void zees_thug_hotline( special_effect_t& effect )
+    {
+      buff_t* radiance_buff;
+      ire_of_devotion_t( const special_effect_t& e, const spell_data_t* spell, const spell_data_t* data, buff_t* rb )
+        : generic_proc_t( e, "ire_of_devotion", spell ),
+          radiance_buff( nullptr )
+      {
+        radiance_buff = rb;
+        base_dd_min = base_dd_max = data->effectN( 3 ).average( e.item );
+      }
+
+      double composite_da_multiplier( const action_state_t* s ) const override
+      {
+        double m = generic_proc_t::composite_da_multiplier( s );
+
+        if ( radiance_buff->check() )
+            m *= 2;
+
+        return m;
+      }
+    };
+
+    tome_of_lights_devotion_cb_t( const special_effect_t& e, buff_t* rb )
+      : dbc_proc_callback_t( e.player, e ),
+        resilience_phase( true ),
+        resilience_verses_tracker( nullptr ),
+        radiance_verses_tracker( nullptr ),
+        inner_radiance_buff( nullptr ),
+        inner_resilience_buff( nullptr ),
+        radiance_buff( nullptr )
+      {
+        radiance_buff = rb;
+
+        inner_radiance_buff = create_buff<stat_buff_with_multiplier_t>( e.player, e.player->find_spell( 450720 ) )
+        ->set_stat_from_effect_type( A_MOD_RATING, e.player->find_spell( 443533 )->effectN( 2 ).average( e.item ) )
+        ->set_rppm( RPPM_DISABLE )
+        ->set_quiet( false );
+
+        inner_resilience_buff = create_buff<stat_buff_with_multiplier_t>( e.player, e.player->find_spell( 450706 ) )
+        ->add_stat(STAT_BONUS_ARMOR, e.player->find_spell( 443533 )->effectN( 4 ).average( e.item ) )
+        ->set_rppm( RPPM_DISABLE )
+        ->set_quiet( false );
+
+
+        resilience_verses_tracker = create_buff<buff_t>( e.player, e.player->find_spell( 450696 ) );
+        resilience_verses_tracker->expire_at_max_stack = true;
+        resilience_verses_tracker->set_expire_callback( [ this ]( buff_t* b, int, timespan_t ) {
+                                      resilience_phase = false;
+                                      inner_resilience_buff->expire();
+                                      inner_radiance_buff->trigger();
+                                      radiance_verses_tracker->trigger();
+                                      } );
+
+        radiance_verses_tracker = create_buff<buff_t>( e.player, e.player->find_spell( 450699 ) );
+        radiance_verses_tracker->expire_at_max_stack = true;
+        radiance_verses_tracker->set_expire_callback( [ this ]( buff_t* b, int, timespan_t ) {
+                                      resilience_phase = true;
+                                      inner_radiance_buff->expire();
+                                      inner_resilience_buff->trigger();
+                                      resilience_verses_tracker->trigger();
+                                      } );
+
+        auto inner_resilience = new special_effect_t( effect.player );
+        inner_resilience->name_str = "inner_resilience";
+        inner_resilience->spell_id = 450706;
```

Comment:
Should probably have some asserts when searching these special effect spell ids just in case they change them for whatever reason.